b2sdk/api.py

```diff
@@ -94,20 +94,23 @@ def __init__(self, account_info=None, cache=None, raw_api=None, max_upload_worke
 
         :param int max_upload_workers: a number of upload threads, default is 10
         """
-        self.raw_api = raw_api or B2RawApi(B2Http())
-        if account_info is None:
-            account_info = SqliteAccountInfo()
-            if cache is None:
-                cache = AuthInfoCache(account_info)
-        self.session = B2Session(self, self.raw_api)
+        self.session = B2Session(account_info=account_info, cache=cache, raw_api=raw_api)
         self.transferer = Transferer(self.session, account_info)
-        self.account_info = account_info
-        if cache is None:
-            cache = DummyCache()
-        self.cache = cache
         self.upload_executor = None
         self.max_workers = max_upload_workers
 
+    @property
+    def account_info(self):
+        return self.session.account_info
+
+    @property
+    def cache(self):
+        return self.session.cache
+
+    @property
+    def raw_api(self):
```

Comment:
I don't think we should be exposing this anymore, since the only thing we need it for is authorize_account which works through session now: https://github.com/reef-technologies/b2-sdk-python/blob/b3b5c2703895c52d17376c9863fa232db2a825f6/b2sdk/api.py#L170

so lets annotate this method for deprecation in the next apiver version